host_modules/docker_service.py

```diff
@@ -4,44 +4,108 @@
 import docker
 import signal
 import errno
+import logging
 
 MOD_NAME = "docker_service"
 
 # The set of allowed containers that can be managed by this service.
-# First element is the image name, second element is the container name.
-ALLOWED_CONTAINERS = [
-    ("docker-syncd-brcm", "syncd"),
-    ("docker-acms", "acms"),
-    ("docker-sonic-gnmi", "gnmi"),
-    ("docker-sonic-telemetry", "telemetry"),
-    ("docker-snmp", "snmp"),
-    ("docker-platform-monitor", "pmon"),
-    ("docker-lldp", "lldp"),
-    ("docker-dhcp-relay", "dhcp_relay"),
-    ("docker-router-advertiser", "radv"),
-    ("docker-teamd", "teamd"),
-    ("docker-fpm-frr", "bgp"),
-    ("docker-orchagent", "swss"),
-    ("docker-sonic-restapi", "restapi"),
-    ("docker-eventd", "eventd"),
-    ("docker-database", "database"),
-]
-
-
-def is_allowed_container(container):
+ALLOWED_CONTAINERS = {
+    "bgp",
+    "bmp",
+    "database",
+    "dhcp_relay",
+    "eventd",
+    "gnmi",
+    "lldp",
+    "pmon",
+    "radv",
+    "restapi",
+    "snmp",
+    "swss",
+    "syncd",
+    "teamd",
+    "telemetry",
+}
+
+# The set of allowed images that can be managed by this service.
+ALLOWED_IMAGES = {
+    "docker-database",
+    "docker-dhcp-relay",
+    "docker-eventd",
+    "docker-fpm-frr",
+    "docker-lldp",
+    "docker-orchagent",
+    "docker-platform-monitor",
+    "docker-router-advertiser",
+    "docker-snmp",
+    "docker-sonic-bmp",
+    "docker-sonic-gnmi",
+    "docker-sonic-restapi",
+    "docker-sonic-telemetry",
+    "docker-syncd-brcm",
+    "docker-syncd-cisco",
+    "docker-teamd",
+}
+
+
+def is_allowed_image(image):
     """
-    Check if the container is allowed to be managed by this service.
+    Check if the image is allowed to be managed by this service.
 
     Args:
-        container (str): The container name.
+        image (str): The image name.
 
     Returns:
-        bool: True if the container is allowed, False otherwise.
+        bool: True if the image is allowed, False otherwise.
     """
-    for _, allowed_container in ALLOWED_CONTAINERS:
-        if container == allowed_container:
-            return True
-    return False
+    image_name = image.split(":")[0]  # Remove tag if present
+    return image_name in ALLOWED_IMAGES
+
+
+def get_sonic_container(container_id):
+    """
+    Get a Sonic docker container by name. If the container is not a Sonic container, raise PermissionError.
+    """
+    client = docker.from_env()
+    if container_id not in ALLOWED_CONTAINERS:
+        raise PermissionError(
+            "Container {} is not allowed to be managed by this service.".format(
+                container_id
+            )
+        )
+    container = client.containers.get(container_id)
+    return container
+
+
+def validate_docker_run_options(kwargs):
```

Comment:
>validate_docker_run_options [](http://example.com/codeflow?start=4&length=27)

Consider restricting `--net=host` option

Reply:
left open in ADO 27057586. Need more discussion here.